Source/WebCore/style/StyleBuilderState.h

```diff
@@ -106,6 +106,7 @@ class BuilderState {
 
     void setIsBuildingKeyframeStyle() { m_isBuildingKeyframeStyle = true; }
 
+    bool isAuthorOrigin() const { return m_currentProperty && m_currentProperty->cascadeLevel == CascadeLevel::Author; };
```

Comment:
@anttijk Would appreciate your thoughts on the approach to detect whether the author has explicitly set `color`.

@rr-codes also needs detect this case to fix another bug, so this will be useful.